### PR TITLE
Implement tapInto

### PR DESCRIPTION
**Pull request by indrekj**
/cc @urmastalimaa

### CODE REVIEW

File: src/Network/Freddy.hs [hunk 1]
```diff
@@ -116,6 +119,26 @@ respondTo connection queueName callback = do
...
+  }
+  AMQP.declareExchange channel AMQP.newExchange {
+    AMQP.exchangeName = topicExchange,
+    AMQP.exchangeType = "topic",
+    AMQP.exchangeDurable = False
+  }
+  AMQP.bindQueue channel "" topicExchange queueName
+
+  tag <- AMQP.consumeMsgs channel queueName AMQP.NoAck $ \(msg, _) ->
+    callback . AMQP.msgBody $ msg
```
Comment (urmastalimaa) [inline src/Network/Freddy.hs]:
``` hs
tag <- AMQP.consumeMsgs channel queueName AMQP.NoAck consumer
  where consumer = callback . AMQP.msgBody . fst
```

**Comment by urmastalimaa** [comment]
Minor comment, +1

File: src/Network/Freddy.hs [hunk 2]
```diff
@@ -134,8 +134,8 @@ tapInto connection queueName callback = do
   }
   AMQP.bindQueue channel "" topicExchange queueName
 
-  tag <- AMQP.consumeMsgs channel queueName AMQP.NoAck $ \(msg, _) ->
-    callback . AMQP.msgBody $ msg
+  let consumer = callback . AMQP.msgBody .fst
+  tag <- AMQP.consumeMsgs channel queueName AMQP.NoAck consumer
```
Comment (indrekj) [inline src/Network/Freddy.hs:138]: @urmastalimaa where syntax gave me a syntax error

**Comment by indrekj** [comment]
!squash

**Pull request merged**